src/s2wasm.h

```diff
@@ -1193,6 +1198,42 @@ class S2WasmBuilder {
         }
       }
```

Comment:
Let me try to be clearer. This line, and the big multiline change under it - lines 1201 to 1236 - are completely unclear to me. Why do they exist? What to they do?